server/project/project.go

```diff
@@ -404,29 +404,34 @@ func (s *Server) GetSyncWindowsState(ctx context.Context, q *project.SyncWindows
 	return res, nil
 }
 
-func (s *Server) NormalizeProjs() {
-	ctx := context.Background()
-	projList, err := s.List(ctx, &project.ProjectQuery{})
+func (s *Server) NormalizeProjs() error {
+	projList, err := s.appclientset.ArgoprojV1alpha1().AppProjects(s.ns).List(metav1.ListOptions{})
 	if err != nil {
-		return
+		return status.Errorf(codes.Internal, "Error retrieving project list: %s", err.Error())
 	}
 	for _, proj := range projList.Items {
-		if proj.NormalizeJWTTokens() {
-			// if !apierr.IsConflict(err), retry 3 times
-			for i := 0; i < 3; i++ {
-				_, err := s.appclientset.ArgoprojV1alpha1().AppProjects(s.ns).Update(&proj)
+		// if !apierr.IsConflict(err), retry 3 times
+		for i := 0; i < 3; i++ {
+			projGet, err := s.appclientset.ArgoprojV1alpha1().AppProjects(s.ns).Get(proj.Name, metav1.GetOptions{})
```

Comment:
API server is going to list all projects and then one by one load each project even if it does not have to be normalized. This might slowdown API server startup. 

This can be fixed by moving line 415 to 427